src/menu/rom_patch_info.c

```
@@ -0,0 +1,452 @@
+    if (file_has_extensions(path_get(path), patch_rom_bps_extensions)) {
+        patch_ext_type = PATCH_TYPE_BPS;
+    } else if (file_has_extensions(path_get(path), patch_rom_ips_extensions)) {
+        patch_ext_type = PATCH_TYPE_IPS;
+    } else if (file_has_extensions(path_get(path), patch_rom_aps_extensions)) {
+        patch_ext_type = PATCH_TYPE_APS;
+    } else if (file_has_extensions(path_get(path), patch_rom_ups_extensions)) {
+        patch_ext_type = PATCH_TYPE_UPS;
+    } else if (file_has_extensions(path_get(path), patch_rom_xdelta_extensions)) {
+        patch_ext_type = PATCH_TYPE_XDELTA;
+    } else {
+        return PATCH_ERR_UNSUPPORTED;
```

Cppcheck (2.20.0)

[error] Resource leak

(resourceLeak)